src/verify/hooks.test.tsx

```diff
@@ -0,0 +1,95 @@
+import { act, fireEvent, render } from '@testing-library/react-native'
+import { FetchMock } from 'jest-fetch-mock/types'
+import React from 'react'
+import { Text } from 'react-native'
+import * as Keychain from 'react-native-keychain'
+import { Provider } from 'react-redux'
+import { phoneNumberRevoked } from 'src/app/actions'
+import Touchable from 'src/components/Touchable'
+import Logger from 'src/utils/Logger'
+import { useRevokeCurrentPhoneNumber } from 'src/verify/hooks'
+import networkConfig from 'src/web3/networkConfig'
+import { createMockStore } from 'test/utils'
+import { mocked } from 'ts-jest/utils'
+
+const mockFetch = fetch as FetchMock
+
+const mockedKeychain = mocked(Keychain)
+mockedKeychain.getGenericPassword.mockResolvedValue({
+  username: 'some username',
+  password: 'someSignedMessage',
+  service: 'some service',
+  storage: 'some string',
+})
+
+const spyLoggerShowMessage = jest.spyOn(Logger, 'showMessage')
+const spyLoggerShowError = jest.spyOn(Logger, 'showError')
+
+function TestComponent() {
+  const revokePhoneNumber = useRevokeCurrentPhoneNumber()
+
+  return (
+    <Touchable onPress={revokePhoneNumber.execute} disabled={revokePhoneNumber.loading}>
+      <Text>Revoke</Text>
+    </Touchable>
+  )
+}
+
+describe('useRevokeCurrentPhoneNumber', () => {
+  it('revokes the phone number', async () => {
+    const store = createMockStore({})
+    const { getByText } = render(
+      <Provider store={store}>
+        <TestComponent />
+      </Provider>
+    )
+    mockFetch.mockResponseOnce(JSON.stringify({ message: 'OK' }), {
+      status: 200,
+    })
+
+    await act(async () => {
```

Comment:
do we need async / await here?

Reply:
Yes, otherwise it doesn't process the underlying work happening.
I've added an explicit `await flushMicrotasksQueue()` to avoid any confusion.